### PR TITLE
Add TIL: Elixir pattern matching for argument equality

### PR DESCRIPTION
**Pull request by maabernethy**

### CODE REVIEW

File: source/posts/2016-05-23-til-elxir-function-header-pattern-matching.md [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+published: true
+tags: til, elixir
+---
+
+[Patterning matching][pattern-matching] and [guard expressions][guard-expressions] are fundamental to
+writing recursive function definitions in [elixir][elixir]. Sometimes guard clauses and pattern matching
+can be used for the same purpose. For Example:
+
+```elixir
+// pattern matching
```
Comment (duggiefresh) [inline source/posts/2016-05-23-til-elxir-function-header-pattern-matching.md]: Should we make these Elixir comments?

 ```elixir
# pattern matching 
```

File: source/posts/2016-05-23-til-elxir-function-header-pattern-matching.md [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+author: Marin Abernethy
+twitter: "pairinMarin"
+github: maabernethy
+published: true
+tags: til, elixir
+---
+
+[Patterning matching][pattern-matching] and [guard expressions][guard-expressions] are fundamental to
+writing recursive function definitions in [elixir][elixir]. Sometimes guard clauses and pattern matching
+can be used for the same purpose. For Example:
```
Comment (duggiefresh) [inline source/posts/2016-05-23-til-elxir-function-header-pattern-matching.md]: No capitalization on "Example"... unless that was a stylistic choice. ;)

File: source/posts/2016-05-23-til-elxir-function-header-pattern-matching.md [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+```elixir
+// pattern matching
+defmodule Exponent do
+  def power(value, 0), do: 1
+  def power(value, n), do: value*power(value, n-1)
```
Comment (duggiefresh) [inline source/posts/2016-05-23-til-elxir-function-header-pattern-matching.md]: Mmm. Our styleguide suggests putting spaces around operators - https://github.com/DockYard/styleguides/blob/master/engineering/elixir.md#whitespace

File: source/posts/2016-05-23-til-elxir-function-header-pattern-matching.md [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+end
+
+// guard expression
+defmodule Exponent do
+  def power(value, n) when n==0, do: 1
```
Comment (duggiefresh) [inline source/posts/2016-05-23-til-elxir-function-header-pattern-matching.md]: The spaces, the spaces

**Comment by duggiefresh** [comment]
Minor comments, LGTD

**Comment by maabernethy** [comment]
Updated per comments! I will need to change the date and file name when this is ready to be merged

File: source/posts/2016-05-23-til-elxir-function-header-pattern-matching.md [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+# guard expression
+defmodule Exponent do
+  def power(value, n) when n == 0, do: 1
+  def power(value, n), do: value * power(value, n - 1)
+end
+```
+
+In both cases above, we only want the first `power` function to run when the second argument is equal to `0`.
+When it is as simple as equality, I tend to use the pattern matching syntax. I typically leave the guard for more complex
+logic like `when rem(x, divisor) == 0`. However, when I need to check whether one argument is equal to another I always used a guard: `when a == b`.
```
Comment (rsocci) [inline source/posts/2016-05-23-til-elxir-function-header-pattern-matching.md]: Maybe reword to "However, when needing to check whether one...". For some reason the way it is now, my brain expects "used" to be in present tense.

File: source/posts/2016-05-23-til-elxir-function-header-pattern-matching.md [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+But, **today I learned**, this can also be handled with pattern matching instead, like so:
```
Comment (rsocci) [inline source/posts/2016-05-23-til-elxir-function-header-pattern-matching.md]: can remove `instead`

**Comment by rsocci** [comment]
lgtm :)

File: source/posts/2016-05-26-til-elxir-pattern-matching-for-argument-equality.md [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+summary: "pattern matching versus guard expressions"
+social: true
+author: Marin Abernethy
+twitter: "pairinMarin"
+github: maabernethy
+published: true
+tags: til, elixir
+---
+
+[Patterning matching][pattern-matching] and [guard expressions][guard-expressions] are fundamental to
```
Comment (poteto) [inline source/posts/2016-05-26-til-elxir-pattern-matching-for-argument-equality.md]: `s/Patterning/Pattern`

File: source/posts/2016-05-26-til-elxir-pattern-matching-for-argument-equality.md [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+writing recursive function definitions in [elixir][elixir]. Sometimes guard clauses and pattern matching
```
Comment (poteto) [inline source/posts/2016-05-26-til-elxir-pattern-matching-for-argument-equality.md]: `s/elixir/Elixir`
